gen3analysis/routes/survival.py

```diff
@@ -1,241 +1,259 @@
+                }
+            )
 
-    return {
-        "meta": {"id": id(result)},
-        "donors": [
-            prepare_donor(donor, interval.get("cumulativeSurvival"))
-            for interval in items
-            for donor in interval["donors"]
-        ],
-    }
+    return pd.DataFrame(records)
 
 
-def get_curve(
-    filters,
-    gdc_graphql_client,
-    req_opts: Optional[Dict] = None,
+async def get_curve(
+    filters, gen3_graphql_client, auth, req_opts: Optional[Dict] = None
 ):
-    filters = [
-        filters,
-        {
-            "op": "or",
-            "content": [
-                {
-                    "op": "and",
-                    "content": [
-                        {
-                            "op": ">",
-                            "content": {
-                                "field": "demographic.days_to_death",
-                                "value": 0,
+    queryFilter = {
+        "and": [
+            filters,
+            {
+                "and": [
+                    {
+                        "or": [
+                            {
+                                "nested": {
+                                    ">": {"days_to_death": 0},
+                                    "path": "demographic",
+                                }
                             },
-                        },
-                    ],
-                },
-                {
-                    "op": "and",
-                    "content": [
-                        {
-                            "op": ">",
-                            "content": {
-                                "field": "diagnoses.days_to_last_follow_up",
-                                "value": 0,
+                            {
+                                "nested": {
+                                    ">": {"days_to_last_follow_up": 0},
+                                    "path": "diagnoses",
+                                }
                             },
-                        },
-                    ],
-                },
-            ],
-        },
-        {"op": "not", "content": {"field": "demographic.vital_status"}},
-        {
-            "op": "in",
-            "content": {
-                "field": "cases.project.project_id",
-                "value": ["MMRF-COMMPASS"],
+                        ]
+                    }
+                ]
             },
-        },
-    ]
-
-    filters = [f for f in filters if "op" in f and "content" in f]
-    data = gdc_graphql_client.execute(
-        query=GDCSurvivalQuery, variables={"filters": {"op": "and", "content": filters}}
+        ]
+    }
+    data = await gen3_graphql_client.execute(
+        access_token=(await auth.get_access_token()),
+        query=Gen3GraphQLQuery,
+        variables={"filter": queryFilter},
     )
-
     if data.get("error"):
-        raise InternalError(data.get("error"))
+        raise ValueError(data.get("error"))
```

Comment:
you shouldn't have to check the error here anymore, i added it to the `gen3_graphql_client.execute` function (although i added "errors" not "error")

Reply:
I removed it.